### PR TITLE
golang: Update to 1.18 from 1.18rc1

### PR DESCRIPTION
**Pull request by dims**
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### What type of PR is this?

/kind feature
/area dependency

#### What this PR does / why we need it:

We cannot ship rc1 ... code freeze is a week away. Let's switch to the 1.18 final release.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Kubernetes 1.24 bumped version of golang it is compiled with to go1.18, which introduced significant changes to its garbage collection algorithm. As a result, we observed an increase in memory usage for kube-apiserver in larger an heavily loaded clusters up to ~25% (with the benefit of API call latencies drop by up to 10x on 99th percentiles). If the memory increase is not acceptable for you you can mitigate by setting GOGC env variable (for our tests using GOGC=63 brings memory usage back to original value, although the exact value may depend on usage patterns on your cluster).
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
/assign @justaugustus @saschagrunert 

follow up from https://github.com/kubernetes/kubernetes/pull/107105 which updated k/k to rc1 and related to https://github.com/kubernetes/release/issues/2307

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -480,7 +480,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(GOFLAGS='' go version)"
   local minimum_go_version
-  minimum_go_version=go1.17.0
+  minimum_go_version=go1.18.0
```
[the remark this answers is not in the document]
Comment (dims) [reply hack/lib/golang.sh:483]: gotcha

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by wojtek-t** [comment]
@dims - I updated the release note in this PR according to the discussion in https://github.com/kubernetes/kubernetes/issues/108357

I'm resolving the corresponding bug based on that.

@liggitt - FYI

**Comment by dims** [comment]
Thanks @wojtek-t

**Pull request merged**